Add broadcom-wl driver for older intel macs

discretizer opened the pull request:
This PR adds support for the broadcom-wl driver.  By default it is disabled but can be enabled by modfiying the driver blacklist files like so: 

```
mv /etc/modprobe.d/blacklist-wl.conf to /etc/modprobe.d/blacklist-wl.conf.disabled
mv /etc/modprobe.d/macbook-wifi.conf.disabled /etc/modprobe.d/macbook-wifi.conf
batocera-save-overlay
```

Additionally to get things to work a patch to `wpa_supplicant`  is needed.  The patch in this PR is taken from the following wpa_supplicant development thread:

https://lists.infradead.org/pipermail/hostap/2025-May/043475.html

This should fix an issue with the broadcom wl driver where AP scans don't work. This is a combination of two separate patches that are already in the upstream git repo, but not in the released version. 

This PR resolves the following issue:
https://github.com/batocera-linux/batocera.linux/issues/7894

And replaces the older PR here: 
https://github.com/batocera-linux/batocera.linux/pull/7735

To revert these changes just reverse the above renames 
Tested on a  Macmini7,1

dmanlfc committed:
add broadcom-wl driver for older macbooks

Add patches for later kernel versions

Add patch for 6.15 kernel

Commit last fix

discretizer commented:
> This was confirmed as not working by multiple Mac users. I don't see what's changed...

As stated... the `wpa_supplicant` step I just updated....

discretizer commented:
If there's a better way to handle this configuration quirk let me know.  I'm fine if the eventual answer is "will not merge" (the driver is ancient anyway),  I just want to get some documentation out there for how to make this work.

discretizer commented:
Ok...  there was ANOTHER WPA supplicant bug.  I made a PR into the buildroot repo to fix that one: 

https://github.com/batocera-linux/buildroot/pulls

discretizer commented:
> The dbus change cannot be added as is. It will affect all users. Also we don't use systemd.

DBus change is not needed anymore with the `wpa_supplicant` PR

dmanlfc commented:
The previous dbus change referenced systemd.
The patch may be ok, I will need to read it in more detail.
Where is the source of this patch? So I can read more about it.

discretizer committed:
Add wpa_supplicant patch

discretizer commented:
Moved the `wpa_supplicant` patch here in proper location per your comment on the `buildroot` PR (https://github.com/batocera-linux/buildroot/pull/1054) and closed that PR.  Patch originates from `wpa_supplicant` development thread here: 
https://lists.infradead.org/pipermail/hostap/2025-May/043475.html

Reverted build root changes in branch.

discretizer committed:
Add newline to patch to fix warning

markmorris commented:
Yeah I’ve decided to do this with a few old Mac minis and none of the WiFi works… can we get a simple implementation please or merge this. I’m not 100 sure how to use this to get it working

dmanlfc reviewed (commented):
Have you tested against kernel 6.18

lbrpdx commented:
What is the status on this? Please be aware that none of the devs has access to every piece of hardware that can run Batocera. Things like old iMacs **need** your help to validate the changes.
